include.sh

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/bash
+
+# Include the slurm utility functions
+source "${BASH_SOURCE[0]%/*}/slurm_utils.sh"
+
+# Do NOT touch these and make sure they are the same as local environment
+# variables!! Otherwise, there can be *duplicate* locations and local containers
+# will not see the same as gitlab containers!
+export CI_WS=${CUSTOM_ENV_CI_WS}
+export LOGFILE="${CI_WS}/gitlab-runner-enroot.log"
+export ENROOT_CACHE_PATH=${CI_WS}/ENROOT_CACHE
+export ENROOT_DATA_PATH=${CI_WS}/ENROOT_DATA
+export SLURM_IDS_PATH=${CI_WS}/SLURM_IDS
+
+
+# Set a variable CONTAINER_NAME based on the environment variable
+# CUSTOM_ENV_USE_NAME
+if [[ -z "${CUSTOM_ENV_USE_NAME}" ]]; then
+    CONTAINER_NAME="GitLabRunnerCustomExecutorEnrootBuildID${CUSTOM_ENV_CI_BUILD_ID}"
+else
+    CONTAINER_NAME="${CUSTOM_ENV_USE_NAME}"
+fi
+export CONTAINER_NAME
+
+
+# CCACHE and volume management
+export CCACHE_DIR_SRC="${CI_WS}/ccache"
+ENROOT_MOUNT_OPTIONS=()
+if [[ -n "${CUSTOM_ENV_CCACHE_DIR}" ]]; then
+    if [[ ! -d "${CCACHE_DIR_SRC}" ]]; then
+        mkdir -p "${CCACHE_DIR_SRC}"
+    fi
+    ENROOT_MOUNT_OPTIONS+=(--mount "${CCACHE_DIR_SRC}:${CUSTOM_ENV_CCACHE_DIR}" )
+fi
+export ENROOT_MOUNT_OPTIONS
+
+
+# Propagate these environment variables to the container
+PROPAGATED_ENV_VARIABLES=(BENCHMARK
```

Comment:
Is this part only for ginkgo?

Reply:
Except for the last 3, yes. At some point, I would like to use another way of setting these variables, either using a file or passing this list in some variable. I will create an issue for cleaning this up.